programs/clearing_house/src/controller/orders.rs

```diff
@@ -2922,12 +2971,26 @@ pub fn fulfill_spot_order_with_match(
         maker_price,
         taker_base_asset_amount,
         base_market.decimals as u32,
+        maker_direction,
     )?;
 
     if base_asset_amount == 0 {
         return Ok(0_u64);
     }
 
+    validate_taker_fill_price(
+        quote_asset_amount,
+        base_asset_amount,
+        taker_direction,
+        taker_price,
+    )?;
+    validate_fill_price(
```

Comment:
should we just have one function `validate_fill_price` pass in taker/maker as a boolean